backend/database.js

```diff
@@ -186,18 +178,22 @@ export default async () => {
     })
 
     sbp('sbp/selectors/overwrite', {
-      'chelonia/db/get': async function (key: string): Promise<Buffer | string | void> {
-        const lookupValue = cache.get(key)
+      'chelonia/db/get': async function (prefixableKey: string): Promise<Buffer | string | void> {
+        const lookupValue = cache.get(prefixableKey)
         if (lookupValue !== undefined) {
           return lookupValue
         }
-        const value = await readData(key)
-        if (value !== undefined) {
-          cache.set(key, value)
+        const [prefix, key] = parsePrefixableKey(prefixableKey)
+        let value = await readData(key)
+        if (value === undefined) {
+          return
         }
+        value = prefixHandlers[prefix](value)
+        cache.set(prefixableKey, value)
         return value
       },
       'chelonia/db/set': async function (key: string, value: Buffer | string): Promise<void> {
+        checkKey(key)
```

Comment:
I'm not sure about putting this here. For example, for the `readData` in `fs` the `checkKey` is still in `readData`, so that's an inconsistency, but more to the point, by keeping it "closer to the action" we can be more sure that it was called before the actual setting happens.